sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/SequencePointCollector.java

```diff
@@ -59,9 +59,17 @@ private static void publishCoverage(Coverage coverage, String filePath, List<Seq
         int line = lineSequencePoints.getKey();
 
         List<SequencePoint> linePoints = lineSequencePoints.getValue();
-        int coveredPoints = (int)linePoints.stream().filter(point -> point.getHits() > 0).count();
 
-        coverage.addBranchCoverage(filePath, new BranchCoverage(line, linePoints.size(), coveredPoints));
+        Map<String, List<SequencePoint>> groupsByCoverageKey = linePoints.stream().collect(Collectors.groupingBy(SequencePoint::getCoverageKey));
+
+        int coveredPoints = 0;
+        for (Map.Entry<String, List<SequencePoint>> group: groupsByCoverageKey.entrySet()){
```

Comment:
maybe rename `group` to `sequencePointEntries` - it's basically the various entries for the same sequence point

Reply:
I think in this case we can let it as it is, since it's used close to declaration and keeps the lines short.